pyvista/core/filters.py

```diff
@@ -1646,21 +1675,16 @@ def plot_over_line(dataset, pointa, pointb, resolution=None, scalars=None,
         except ImportError:
             raise ImportError('matplotlib must be available to use this filter.')
 
-        if resolution is None:
-            resolution = int(dataset.n_cells)
-        if not isinstance(resolution, int) or resolution < 0:
-            raise RuntimeError('`resolution` must be a positive integer, not {}'.format(type(resolution)))
-        # Make a line and probe the dataset
-        line = pyvista.Line(pointa, pointb, resolution=resolution)
-        sampled = line.sample(dataset)
+        # Sample on line
+        sampled = DataSetFilters.sample_over_line(dataset, pointa, pointb, resolution)
```

Comment:
This must be dataset.sample_...

More generally maybe there should be a static analyser on CI to catch such things.

Reply:
I've implemented a lot of filters that call other filters this way. I thought it was "safer"

Reply:
is this bad practice, @imsodin?

Reply:
This whole class is littered with both types examples.  What is the preferred approach, and I can use that one?

Reply:
I'm happy with it as is, but yes, you're right that the class has both examples all over the place. We're working to refactor things like this to make them more consistent on a separate branch, so I say just leave it and we'll choose the "proper" method later

Reply:
Ah I haven't been aware this was used in the same file. Maybe it's just simply lack of experience on my part that i have never seen a class instance method be called like this. The only practice I could come up with that might be somewhat relevant is that in python there should be "one, preferably obvious way of doing things" and if I had to choose I'd definitely choose variable.method(... over class.method(variable, ...

Reply:
I agree with @imsodin .  I found this [reference](https://docs.python.org/3/tutorial/classes.html#inheritance) that 

> There is a simple way to call the base class method directly: just call BaseClassName.methodname(self, arguments).

This is the strategy used here.  I see this being useful if you don't want derived classes to overwrite the sample_over_line method call inside the plot_over_line method, but you do want to be able to overwrite it in general.

I'm not sure this is what you want however.

Reply:
Regardless of whether or not it should change, that's a topic for a separate PR.